storage/blockdevice/COMPONENT_SPINAND/include/SPINAND/bch.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Generic binary BCH encoding/decoding library
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published by
```

Comment:
I need to check why license check did not fail. This should trigger the license-check failure

The file can't be accepted with the license provided unfortunately (not permissive).

Reply:
OK! I will fix this issue about BCH license.

Reply:
> I need to check why license check did not fail. This should trigger the license-check failure
> 
> The file can't be accepted with the license provided unfortunately (not permissive).

I have update BCH code and license, please review it.